ddtrace/internal/writer.py

```diff
@@ -435,6 +443,9 @@ def write(self, spans=None):
             )
             self._metrics_dist("buffer.dropped.traces", 1, tags=["reason:full"])
             self._metrics_dist("buffer.dropped.bytes", payload_size, tags=["reason:full"])
+        except Exception:
+            log.error("failed to add spans to the encoder buffer%r", self._encoder, exc_info=True)
+            self._metrics_dist("buffer.dropped.traces", 1, tags=["reason:unknown"])
```

Comment:
instead of `unknown` would it make sense to have as `reason:encoding` here?

if we can get some more specific exceptions from `encoder.put()` we might be able to have some additional reasons which would be nice

Reply:
oh, I guess this is for "buffer.dropped" so "unknown" might be ok here, or maybe we can do the name of the Exception class?

e.g. `reason:typeerror`, `reason:valueerror`, etc ?

Reply:
It's a bit jank but I am using this `"reason:{}".format(e.__class__.__name__.lower())` to get the error tag to the name of the exception class

Reply:
yeah, `e.__class__.__name__.lower()` is fine

only other alternative is `type(e).__name__.lower()` which is basically the same thing, so no issues with what you have